.github/pull_request_template.md

```diff
@@ -4,30 +4,12 @@ Fixes #[insert GH or internal issue number(s)].
 
 ## Author has addressed the following
 
-- Tests
-  - [ ] TODO (before merge)
-  - [ ] Tests included
-  - [ ] Tests not necessary because:
-- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
-  - [ ] I don't know
-  - [ ] Required
-  - [ ] Not required because:
-- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
-  - [ ] TODO (before merge)
-  - [ ] Changeset included
-  - [ ] Changeset not necessary because:
-- Public documentation
-  - [ ] TODO (before merge)
-  - [ ] Cloudflare docs PR(s): <!--e.g. <https://github.com/cloudflare/cloudflare-docs/pull/>...-->
-  - [ ] Documentation not necessary because:
+- [ ] Tests
+- [ ] E2E Tests (Use "e2e" label to run them on CI, or ask maintainer to run separately)
+- [ ] [Changeset](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets)
+- [ ] Docs
 
 <!--
 Have you read our [Contributing guide](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md)?
-In particular, for non-trivial changes, please always engage on the issue or create a discussion or feature request issue first before writing your code.
```

Comment:
People (like me) might use the PR itsel f as an engagement point (and we should actually encourage it).

Reply:
One thing I'd be slightly concerned about with this change is an external contributor opening up a PR for a feature we don't want to add for whatever reason, and then us having to close the PR after they've already done a bunch of work (as opposed to closing an issue after just a suggestion of a feature). There probably won't be loads of cases of that though, to be fair, and maybe that's an okay tradeoff?

Reply:
Yup, that's what I think.

Reply:
I suspect what would actually happen in practice, rather than the external contributor going and creating an issue for discussion first, they would spend ages making the PR, then after pushing it realise that they need an issue, go create the issue, and then we have to close both the PR and issue if it is something we didn't want in the first place.

Reply:
To clarify, that means you're in support of removing this line/requirement right? Because you're describing the situation as it exists right now?

(I also want to make sure we're _not_ optimising to prevent the rare scenario of an external contributor making a big PR without ever talking to us. Has it happened often?)

Reply:
for some context here, the reason this was added is for exactly the scenario @penalosa cited -- we were finding ourselves having to close PRs that implemented features that hadn't been discussed with the team, but which didn't make sense to include for various reasons. the goal is to save folks from doing work that won't be landable. 

i agree that at the point of someone seeing this it's likely too late to save them doing the work, but i think having some language to signal process can 1) help manage expectations if the PR does end up being closed, and 2) reduce potential future friction

(as a note, while this is specified in the contributing guide [here](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md?plain=1#L9), it can be easily missed)